### PR TITLE
Feature/#117/smart car collision damage

### PR DESCRIPTION
**Pull request by Q16solver**

### CODE REVIEW

File: Assets/Gameplay/Car/CarManager.cs [hunk 1]
```diff
@@ -54,15 +57,23 @@ void Start()
         }
 
         /// <summary>
-        /// Apply damage to car on collision with other players and the environment (e.g. buildings), by decreasing the players
-        /// health by a flat amount.
+        /// Apply damage to car on collision with another player's front of the car, the environment (e.g. buildings)
+        /// or if the collision is between the back of the two cars, by decreasing the players health by an amount proportional 
+        /// to the force of the collision.
         /// </summary>
```
Comment (RFSurdsman) [inline Assets/Gameplay/Car/CarManager.cs]: Change this comment to:
`Damage our self when hit by another player's front of car, the environment or the back of the other car hits the back of our car, by decreasing the player's health by an amount proportional to the force of the collision. The purpose of this is to minimise the chance of the aggressor car taking damage when ramming into other cars.`

File: Assets/Gameplay/Car/CarManager.cs [hunk 2]
```diff
@@ -70,7 +70,7 @@ void OnCollisionEnter(Collision collision)
             bool isHitByOtherCarFront = contactPoint.otherCollider.gameObject.CompareTag(Tags.CarFrontCollider);
             bool isHitByOtherCarBackIntoOurBack = contactPoint.thisCollider.gameObject.CompareTag(Tags.CarBackCollider) && contactPoint.otherCollider.gameObject.CompareTag(Tags.CarBackCollider);
             
-            if (isHitByEnvironment | isHitByOtherCarFront || isHitByOtherCarBackIntoOurBack)
+            if (isHitByEnvironment || isHitByOtherCarFront || isHitByOtherCarBackIntoOurBack)
```
Comment (RFSurdsman) [inline Assets/Gameplay/Car/CarManager.cs]: Xd gg

**Comment by RFSurdsman** [comment]
Build the game and push it to `unity-bin` so we can try it out!

File: Assets/Gameplay/Car/CarManager.cs (hunk 1 above)
Comment (Q16solver) [reply Assets/Gameplay/Car/CarManager.cs]: Changed.

**Pull request merged**